src/onnx/conv.cpp

```diff
@@ -47,6 +49,34 @@ void recalc_conv_attributes(value& v, size_t kdims)
     }
 }
 
+static instruction_ref apply_nhwc_perm(const onnx_parser::node_info& info,
+                                       instruction_ref ins,
+                                       const std::function<void(std::vector<int>&)>& f)
+{
+    std::vector<int> perm(ins->get_shape().ndim());
+    std::iota(begin(perm), end(perm), 0);
+    f(perm);
+    return info.add_instruction(make_op("transpose", {{"permutation", perm}}), ins);
+}
```

Comment:
Using a HOF seems overkill. There is just one permutation and to go the other way you can use `invert_permutation` to derive it, so a simple flag will be simpler:

```cpp
static instruction_ref apply_nhwc_perm(const onnx_parser::node_info& info,
                                       instruction_ref ins,
                                       bool invert)
{
    std::vector<int64_t> perm(ins->get_shape().ndim());
    std::iota(perm.begin() + 1, perm.end() - 1, 2);
    perm.back() = 1;
    return info.add_instruction(make_op("transpose", {{"permutation", invert ? invert_permutation(perm) : perm}}), ins);
}
```

Reply:
Done.